libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -416,4 +427,66 @@ void AP_Camera_Backend::log_picture()
 }
 #endif
 
+#if AP_CAMERA_TRACKING_ENABLED
+// set tracking to none, point or rectangle (see TrackingType enum)
+// if POINT only then top_left is the point
+// top_left,bottom_right are in range 0 to 1.  0 is left or top, 1 is right or bottom
+bool AP_Camera_Backend::set_tracking_external(TrackingType tracking_type, const Vector2f& top_left, const Vector2f& bottom_right)
+{
+    if (tracker == nullptr) {
+        return false;
+    }
+    return tracker->set_tracking(tracking_type, top_left, bottom_right, _params.track_sysid, _params.track_compid, camera_settings._cam_info);
+}
+
+// start object tracking
+bool AP_Camera_Backend::set_tracking(TrackingType tracking_type, const Vector2f& top_left, const Vector2f& bottom_right)
+{
+    if (_params.track_enable == 1) {
+        return set_tracking_external(tracking_type, top_left, bottom_right);
+    }
+    return set_tracking_internal(tracking_type, top_left, bottom_right);
+}
+#endif
+
+// handle camera information message
+void AP_Camera_Backend::handle_message_camera_information(mavlink_channel_t chan, const mavlink_message_t &msg)
+{
+    // accept the camera information only if its coming either from camera or from external tracking system
+    if (msg.sysid != _params.track_sysid || msg.compid != _params.track_compid) {
+        return;
+    }
+
+    mavlink_msg_camera_information_decode(&msg, &camera_settings._cam_info);
+
+    const uint8_t fw_ver_major = camera_settings._cam_info.firmware_version & 0x000000FF;
+    const uint8_t fw_ver_minor = (camera_settings._cam_info.firmware_version & 0x0000FF00) >> 8;
+    const uint8_t fw_ver_revision = (camera_settings._cam_info.firmware_version & 0x00FF0000) >> 16;
+    const uint8_t fw_ver_build = (camera_settings._cam_info.firmware_version & 0xFF000000) >> 24;
+
+    // display camera info to user
+    gcs().send_text(MAV_SEVERITY_INFO, "Camera: %s.32 %s.32 fw:%u.%u.%u.%u",
```

Comment:
This should be `%.32s` instead, probably carried over from the code before #27562